sdk/core/Azure.Core.Experimental/src/JsonData.cs

```diff
@@ -0,0 +1,1090 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Dynamic;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// A mutable representation of a JSON value.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class JsonData : IDynamicMetaObjectProvider
+    {
+        private readonly JsonValueKind _kind;
+        private Dictionary<string, JsonData>? _objectRepresentation;
+        private List<JsonData>? _arrayRepresentation;
+        private object? _value;
+
+        private static readonly JsonSerializerOptions DefaultJsonSerializerOptions = new JsonSerializerOptions();
+
+        /// <summary>
+        ///  Creates a new JsonData object which represents an JSON object with no properties.
+        /// </summary>
+        public JsonData() : this(System.Array.Empty<KeyValuePair<string, JsonData>>())
+        {
+        }
+
+        /// <summary>
+        ///  Creates a new JsonData object which represents the value of the given JsonDocument.
+        /// </summary>
+        /// <param name="jsonDocument">The JsonDocument to convert.</param>
+        /// <remarks>A JsonDocument can be constructed from a JSON string using <see cref="JsonDocument.Parse(string, JsonDocumentOptions)"/>.</remarks>
+        public JsonData(JsonDocument jsonDocument) : this((object?)jsonDocument)
```

Comment:
Wanted to have the constructor based on your feedback that customers are more likely to discover constructors than factory methods and this would be the hook that someone would use who had a stringified JSON documented they wanted to send.

Reply:
Is this the overload for people who want to pass in a JSON document string (as we removed string overload)? If yes, let's just merge it as-is for the UX study, but I have to say now that I look at the signature (as opposed to just talking about it) I doubt people will think it's the way to pass JSON payloads if they have the payload as string.

Reply:
> Is this the overload for people who want to pass in a JSON document string (as we removed string overload)? If yes, let's just merge it as-is for the UX study.

Yup.  Sounds good!